client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkUtils.java

```diff
@@ -344,6 +344,20 @@ public static void cancelShuffle(int shuffleId, String reason) {
               .hiddenImpl(TaskSetManager.class, "taskInfos")
               .defaultAlwaysNull()
               .build();
+  private static final DynFields.UnboundField<
+          scala.collection.mutable.HashMap<
+              Integer, scala.collection.mutable.HashMap<Integer, TaskSetManager>>>
+      TASK_SETS_BY_STAGE_ID_AND_ATTEMPT_FIELD =
+          DynFields.builder()
+              .hiddenImpl(TaskSchedulerImpl.class, "taskSetsByStageIdAndAttempt")
```

Comment:
```
  // TaskSetManagers are not thread safe, so any access to one should be synchronized
  // on this class.  Protected by `this`
  private val taskSetsByStageIdAndAttempt = new HashMap[Int, HashMap[Int, TaskSetManager]]
```
 
It looks like this variable is not thread-safe, will this be safe if you get the variable by reflection?

Reply:
Thanks, Have wrapped to code with `synchronized (taskScheduler)`.